assemblers/org.osbuild.qemu

```diff
@@ -52,9 +52,33 @@ SCHEMA = """
     }
   },
   "format": {
-    "description": "Image file format to use",
-    "type": "string",
-    "enum": ["raw", "raw.xz", "qcow2", "vdi", "vmdk", "vpc", "vhdx"]
+    "oneOf": [
+      {
+        "description": "Image file format to use (legacy; options and compression are hardcoded for each format, use the object form)",
+        "type": "string",
+        "enum": ["raw", "raw.xz", "qcow2", "vdi", "vmdk", "vpc", "vhdx"]
+      },
+      {
+        "type": "object",
+        "required": ["type"],
+        "properties": {
+          "type": {
+            "description": "Image file format to pass to qemu-img convert -O",
+            "type": "string",
+            "enum": ["raw", "raw.xz", "qcow2", "vdi", "vmdk", "vpc", "vhdx"]
+          },
+          "options": {
+            "description": "Format options to pass to qemu-img convert -o (ignored for raw and raw.xz)",
+            "type": "string"
+          },
+          "compressed": {
+            "description": "Compress the image using qemu-img convert -c (ignored for raw and raw.xz)",
```

Comment:
Could drop `raw.xz` as option and have `compressed: True` + `type: raw` result in `raw.xz`.

Reply:
I would rather not overload the `compressed` field, it's hard to document. Multi-stage assemblers would be nice for this usecase...